Library/Homebrew/formulary.rb

```diff
@@ -105,7 +105,16 @@ def load_file
   # Loads formulae from bottles.
   class BottleLoader < FormulaLoader
     def initialize(bottle_name)
-      @bottle_filename = Pathname(bottle_name).realpath
+      case bottle_name
+      when %r{(https?|ftp|file)://}
+        @bottle_filename = HOMEBREW_CACHE/File.basename(bottle_name)
+        unless @bottle_filename.exist?
+          HOMEBREW_CACHE.mkpath
+          curl bottle_name, "-o", @bottle_filename
```

Comment:
I think the `CurlBottleDownloadStrategy` should be used here instead to handle e.g. caching and resuming.